src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Component, Input, AfterViewInit, Inject, ElementRef, OnInit
+} from '@angular/core';
+import {DOCUMENT} from '@angular/platform-browser';
+import {Router, ActivatedRoute, Event, NavigationEnd} from '@angular/router';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+
+@Component({
+  selector: 'table-of-contents',
+  styleUrls: ['./table-of-contents.scss'],
+  template: `
+    <div *ngIf="links?.length" class="toc-container">
+      <h2>Contents</h2>
```

Comment:
I believe `<h2>` isn't appropriate here; `h1` - `h6` are meant for content section headings, not really navigation